### DISCUSSION TITLE
Disable Internal nginx Proxy

### DISCUSSION THREAD

**Discussion by Svagtlys** [seanmorley15/AdventureLog#367] [Q&A] [answered]
Is there a way to disable the internal nginx proxy? I've been using caddy to serve the media files (never used the external nginx), and with the update adding the internal nginx I'm getting an error on start saying "Starting nginx: nginx failed!". I redid my compose to be almost exactly the same as what is in the repo (not serving files via caddy, update envs, remove custom user id), but am still having issues.

**Comment by seanmorley15** [comment, marked as answer]
While right now I don't have an exact way to disable it because its the standard feature... that does not mean you have to use it. The main reason why this error is popping up and preventing container start is the nginx conf used by default assumed that there is docker service called "server". If you want to call it "adventurelog-server" or something else, you would have to get a copy of the nginx conf used, found here: https://github.com/seanmorley15/AdventureLog/blob/main/backend/nginx.conf. Then mount it into the container at /etc/ngninx/nginx.conf, changing any reference to "server" with your desired service name. I know you are not exactly using it but this should at least prevent it from preventing startup. If this is still causing issues feel free to send your compose so I can take a look and help ! Sorry this is happening

**Reply by Svagtlys** [reply]
Thanks for the steps on how to fix it, that is indeed what is probably causing my issue (using a different container name).

**Reply by seanmorley15** [reply]
Sounds good! Let me know if you have any other questions!

**Reply by Svagtlys** [reply]
Been doing some work on this and still can't get it to work. Is it every reference to server, or just the second server on line 14 and the one on line 22?

**Reply by seanmorley15** [reply]
These are the references to change in the nginx file.
![image](https://github.com/user-attachments/assets/fb3d9007-aa6b-40bc-85ae-f40054979b34)
The first one does not have http, second one does. Just change `server` with your service name

**Reply by Svagtlys** [reply]
Hm I was struggling with this, so I just ended up removing the http block entirely. Now nginx starts and I can go back to serving my mediafiles via caddy :smile:
